src/remote-data.ts

```diff
@@ -347,6 +372,12 @@ export interface CaseMap<L, A, B> {
   success: Function1<A, B>;
 }
 
+export interface WedgeCaseMap<L, A, B> {
+  none: B;
+  failure: Function1<L, B>;
+  some: Function1<A, B>;
```

Comment:
In the Haskell `Wedge` type, the three constructors are named: `Nowhere`, `Here`, `There`. That's kinda cute, but maybe a little too cute? I feel like the TypeScript community has taken more naming cues from like Scala, F#, and Rust communities, which tend to have less cute names. Any opinions here?

Reply:
Agreed on nowhere/here/there being "too cute".

I'm not sold on `none` vs `some` though; I would assume (if I hadn't read this code) that `none` represented "no elements in the list" and `some` represented "some elements in the list", rather than "not fetched" vs "fetched". Maybe that maps to the FP literature well but I'd be surprised if folks don't find it confusing too.

I don't have any better ideas though...

Reply:
When we use `toOption` we get an `Option` which has two constructors: `None` and `Some`. This wedge datatype has an additional constructor, which in FP land is called `Left` (Haskell, fp-ts, Scala, OCaml), `Err` (Rust), `Failure` (Swift).

In this mapping between `RemoteData` and `Wedge`, we have the following:

![RemoteData to Wedge](https://user-images.githubusercontent.com/1682194/99541342-bab56400-297e-11eb-8007-2c218f0ec0ac.png)

We could call them: `None`, `Failure`, `Success`, if that seems more obvious in usage.

Reply:
Hmm I suppose Some is probably better - "success"  would be come ambiguous in that case which doesn't seem like an improvement either.

I'll trust your judgement here 👍